### PR TITLE
Tweak test projects so the console window remains open

### PR DESCRIPTION
**Pull request by amznestebanpapp**
Currently we have some projects that when run from VS will leave the console window open, but we have some projects that wont. The tests that leave the console open are the googletest ones that are not custom targets.
For pytest and projects that use custom targets the console will close immediately after the execution finishes.
The custom target ones are used when the tests are inside another dll/exe that is already associated to a test.

This injects to all tests a property that treats it as a "Console" project which leaves the console window open. 

Solves Issue #2279

### CODE REVIEW

File: cmake/Platform/Common/TestProject.props [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+-->
+
+<Project>
+    <ItemDefinitionGroup>
+        <Link>
+            <!-- Setting test projects as "Console" so they leave the console open after they finish -->
+            <SubSystem>Console</SubSystem>
+        </Link>
+    </ItemDefinitionGroup>
+</Project>
```
Comment (lumberyard-employee-dm) [inline cmake/Platform/Common/TestProject.props:16]: *cough* no newline *cough*

**Review by lumberyard-employee-dm — APPROVED** [review]

File: cmake/Platform/Common/TestProject.props (hunk 1 above)
Comment (amznestebanpapp) [reply cmake/Platform/Common/TestProject.props:16]: adding

Comment (lumberyard-employee-dm) [reply cmake/Platform/Common/TestProject.props]: > devs are happy now

**Review by lawsonamzn — APPROVED** [review]
Switching an applications subsystem to "Console" does a lot more than _just_ leaving the console open ,but it does do that.

**Pull request merged**